crates/cli/src/main.rs

```diff
@@ -336,6 +338,39 @@ async fn main() -> anyhow::Result<()> {
     }))
     .await;
 
+    let dump_state = config.dump_state.clone();
+    let dump_interval = config
+        .state_interval
+        .map(Duration::from_secs)
+        .unwrap_or(Duration::from_secs(60)); // Default to 60 seconds
+    let preserve_historical_states = config.preserve_historical_states;
+
+    // If user provided a dump state path, spawn the periodic dumper
+    if let Some(dump_path) = dump_state {
+        let node_for_dumper = node.clone();
+        let mut state_dumper = PeriodicStateDumper::new(
+            node_for_dumper,
+            Some(dump_path),
+            dump_interval,
+            preserve_historical_states,
+        );
+
+        // Spawn a task to handle periodic dumping and final dump on shutdown
+        let dumper_handle = tokio::spawn(async move {
+            tokio::select! {
+                _ = signal::ctrl_c() => {
+                    tracing::trace!("received shutdown signal, shutting down");
+                },
+                _ = &mut state_dumper => {}
+            }
+
+            state_dumper.dump().await;
+
+            std::process::exit(0);
```

Comment:
If this is going to be our "exit point" then could you move `BlockProducer` here as well? So it's a part of `select!` and is not spawned on its own.

Reply:
I incorporates this approach in https://github.com/matter-labs/anvil-zksync/pull/497, feel free to rebase on those changes and just add state dumper to the existing `select!` now